### DISCUSSION TITLE
set namespace and come out of namespace

### DISCUSSION THREAD

**Discussion by greylegface** [orchestral/parser#102] [Q&A]
Hi.

I have an issue parsing the following code:

````
<messages>
    <legdtl:legDetail>
        <identifier>
            <data>example</data>
        </identifier>
    </legdtl:legDetail>   
</messages>
````

I have extracted the data and I am parsing it like so:

````
$data = $data->parse([
    'data' => ['uses' => 'legdtl:legDetails.identifier.data']
]);
````

which doesn't work.

I have tried namespacing the tag as well:

````
$data = $data->rebase('messages')->namespaced('legdtl', [
    'data' => ['uses' => 'legDetail.identifier.data']
]);
````

Can anyone point me in the direction of getting out of the namespace, as the tag child doesn't have the namespace as the parent? I would guess that because the <identifier> doesnt have the namespace <legdtl:> is why it cant find it.

**Comment by crynobone** [comment]
Namespace atttributes require valid definition, which I don't see in the example. E.g:

```
<root xmlns:foo="http://www.w3c.org/2001/XMLSchema">

</root>
```

Our parser will first ensure that `http://www.w3c.org/2001/XMLSchema` contain the correct definition for `foo` namespace and the only fetch the content. 

Refer to closed issues for some sample on this.

**Comment by greylegface** [comment]
Hi.

I assumed that was obvious so  I didn't have to type it out.

this is what my XML looks like:

````
<trans:transaction xmlns:trans="http://www.lsb.de/apc/ops/Transaction" xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance" xmlns:legdtl="http://www.lsb.de/apc/ops/legDetail" xsi:schemaLocation="http://www.lsb.de/apc/ops/Transaction ../ops/Transaction.xsd http://www.lsb.de/apc/ops/legDetail ../ops/legDetail.xsd">
<messages>
    <legdtl:legDetail>
        <identifier>
            <data>example</data>
        </identifier>
    </legdtl:legDetail>   
</messages>
</trans:transaction>
````

The problem I have is getting anything after the `legdtl:legDetail` node, as the children aren't namespaced.

Thanks
